extensions/amp-connatix-player/0.1/amp-connatix-player.js

```diff
@@ -71,10 +105,86 @@ export class AmpConnatixPlayer extends AMP.BaseElement {
         // Ignore messages from other iframes.
         return;
       }
-      // Player wants to close because the user interracted on its close button
-      if (getData(e) === 'cnx_close') {
-        this.destroyPlayerFrame_();
-        this.attemptCollapse();
+      const dataString = getData(e);
+      const dataJSON = tryParseJson(dataString) || dict();
+
+      if (dataJSON['event'] !== 'command') {
+        return;
+      }
+
+      switch (dataJSON['func']) {
+        // Player wants to close because the user interacted on its close button
+        case 'cnxClose': {
+          this.destroyPlayerFrame_();
+          this.attemptCollapse();
+          break;
+        }
+        // Player rendered
+        case 'cnxPlayerRendered': {
+          this.playerReadyResolver_(this.iframe_);
+          break;
+        }
+      }
+    });
+  }
+
+  /**
+   * Binds to amp-consent
+   * @private
+   */
+  bindToAmpConsent_() {
+    const consentPolicyId = super.getConsentPolicy() || 'default';
```

Comment:
To confirm: even if a publisher doesn't block the `<amp-connatix-player>` with `data-block-on-consent` attribute, you would still want to have the player blocked by the consent?

Reply:
Hi @zhouyx! No, we don't want to have the player blocked by the consent. When I tested this, I saw on local using Resource Override extension for Chrome the case when super.getConsentPolicy() returns null and in that case, the policy-state was Insufficient. After that, we added || 'default' and the policy-state was Sufficient. 

This was the case for what we did const consentPolicyId = super.getConsentPolicy() || 'default';

In case it can block our component even if a publisher doesn't block the <amp-connatix-player> with data-block-on-consent attribute we will remove || 'default'. Thanks @zhouyx for your review!

Reply:
@zhouyx  I changed from 
const consentPolicyId = super.getConsentPolicy() || 'default';
to
const consentPolicyId = super.getConsentPolicy();
but it fails because super.getConsentPolicy() can return null but consentPolicyId should be (string|undefined)

Travis error here: https://travis-ci.org/github/ampproject/amphtml/jobs/715082797

Which solution would you suggest here? Thanks

Reply:
I'm not familiar with the Resource Override Chrome extension. 
The idea is this. `getConsentPolicyState()` requires a valid id where in most case it will be `'default'`. It returns the doc level consent state regardless of the `<amp-connatix-player>` setting. 

1. If you want to handle the consent blocking logic yourself, instead of having the AMP runtime to block it for you. You may want to override the `getConsentPolicy()` method. Return `null` if you dont want the AMP runtime to block the `<amp-connatix-player>`, otherwise leave the method as it is. 
2. If you don't need binding to consent state when `data-block-on-consent` is not present. I would have something like the following.
```
if (!super.getConsentPolicy()) {
  return
} else {
  bindAmpConsent()
}
```

Reply:
@zhouyx The usage of `const consentPolicyId = super.getConsentPolicy() || 'default'; ` is as needed and work similar as https://github.com/ampproject/amphtml/pull/29636/files.

I'll resolve this conversation. Thank you! Please let me know if there is something else needed in order to merge this PR.